tokenizers/src/models/unigram/model.rs

```diff
@@ -78,18 +81,35 @@ pub enum UnigramError {
 impl Default for Unigram {
     fn default() -> Self {
         let vocab = vec![("<unk>".to_string(), 0.0)];
-        Self::from(vocab, Some(0)).unwrap()
+        Self::from(vocab, Some(0), Some(false)).unwrap()
     }
 }
 
+pub fn byte_to_piece(c: u8) -> String {
+    format!("<0x{:02X}>", c)
+}
+
+pub fn piece_to_byte(piece: &str) -> Option<u8> {
+    let mut k_map: HashMap<String, u8> = HashMap::new();
+    for i in 0..=255 {
+        let byte_piece = byte_to_piece(i);
+        k_map.insert(byte_piece.clone(), i);
+    }
+    k_map.get(piece).copied()
+}
+
 impl Unigram {
     /// Create a `Unigram` model from a given vocabulary.
     /// Vocabulary are the various tokens and their associated score which is a sort of a logprob of
     /// their frequency, which will enable tokenization and sampling.
     /// unk_id, is the index within the vocabulary.
     /// For now `Unigram` *requires* at least `unk` because we might find a never seen char.
     /// Further versions might allow that part to be hidden.
-    pub fn from(vocab: Vec<(String, f64)>, unk_id: Option<usize>) -> Result<Self> {
+    pub fn from(
+        vocab: Vec<(String, f64)>,
+        unk_id: Option<usize>,
+        byte_fallback: Option<bool>,
```

Comment:
Why `Option<bool>` ? It should be `bool` no ?

There are no optional arguments in Rust (and it's good)

`unk_id` is really an Option, it's not at all forced (but it will cause errors if you haven't one and are triggering an unk).

Reply:
I am too pythonic and tried to have a default to False and make it an option

Reply:
@ArthurZucker Tbh I still think pythonically too and I'm wondering what is the idiomatic Rust thing to do here

Reply:
I think that it doesn't make sense to have to init a unigram with `Some(false)` which you have to do rigght now !